File: back-end/src/main/java/com/espto/espto/service/StateService.java
package com.espto.espto.service;

import com.espto.espto.common.GenericService;
import com.espto.espto.domain.State;
import com.espto.espto.repository.StateRepository;
import org.springframework.stereotype.Service;

@Service
public class StateService extends GenericService<State, Long, StateRepository> {

    public StateService(StateRepository repository) {
        super(repository);
    }
}

File: front-end/src/app/demo/service/state-service.ts
import { Injectable } from "@angular/core";
import { CrudService } from "./crud-service";
import { HttpClient } from "@angular/common/http";
import { environment } from "src/environments/environment";

@Injectable({providedIn:'root'})
export class StateService extends CrudService<any> {

    constructor(protected override http: HttpClient) {
        super(http , environment.API + '/state');
     }
}
